### PR TITLE
Ria 5345 bail summary uploaded notification

### PR DESCRIPTION
**Pull request by ThomasKKC**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-5345

### Change description ###
Added notifications for legal rep and admin (bail hearing centre) when home office uploaded bail summary

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/handlers/presubmit/NotificationHandler.java [hunk 1]
```diff
@@ -41,7 +41,7 @@ public NotificationHandler(BiPredicate<PreSubmitCallbackStage, Callback<AsylumCa
     public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
         requireNonNull(callbackStage, "callbackStage must not be null");
         requireNonNull(callback, "callback must not be null");
-        if (callback.getEvent() == Event.SUBMIT_APPLICATION) {
+        if (callback.getEvent() == Event.SUBMIT_APPLICATION || callback.getEvent() == Event.UPLOAD_BAIL_SUMMARY) {
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/handlers/presubmit/NotificationHandler.java:44]: May be if we add next event here, lets create a list of forbidden events here.

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/bail/adminofficer/email/AdminOfficerBailSummaryUploadedPersonalisation.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        this.emailAddressFinder = emailAddressFinder;
+    }
+
+    @Override
+    public String getTemplateId() {
+        return bailSummaryUploadedAdminOfficerTemplateId;
+    }
+
+    @Override
+    public Set<String> getRecipientsList(BailCase bailCase) {
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/bail/adminofficer/email/AdminOfficerBailSummaryUploadedPersonalisation.java:38]: For Admin notification, do we need to use BailHearingCentre Email?

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/bail/adminofficer/email/AdminOfficerBailSummaryUploadedPersonalisation.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @Override
+    public Map<String, String> getPersonalisation(BailCase bailCase) {
+        requireNonNull(bailCase, "bailCase must not be null");
+
+        return ImmutableMap
+                .<String, String>builder()
+                .put("bailReferenceNumber", bailCase.read(BailCaseFieldDefinition.BAIL_REFERENCE_NUMBER, String.class).orElse(""))
+                .put("legalRepReference", bailCase.read(BailCaseFieldDefinition.LEGAL_REP_REFERENCE, String.class).orElse(""))
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/bail/adminofficer/email/AdminOfficerBailSummaryUploadedPersonalisation.java:54]: Check with Elli and Ryan, for scenario when HO/Admin submits Application with no LR, this field will appear blank. I think we need to create this as Optional Content. Ryan is working on something similar on Submit Application event.

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/BailNotificationGeneratorConfiguration.java [hunk 4]
```diff
@@ -30,4 +32,21 @@ public List<BailNotificationGenerator> submitApplicationHearingCentreNotificatio
...
+    public List<BailNotificationGenerator> uploadSummaryNotificationGenerator(
+            AdminOfficerBailSummaryUploadedPersonalisation adminOfficerBailSummaryUploadedPersonalisation,
+            LegalRepresentativeBailSummaryUploadedPersonalisation legalRepresentativeBailSummaryUploadedPersonalisation,
+            NotificationSender notificationSender,
+            BailNotificationIdAppender notificationIdAppender) {
+
+        return Arrays.asList(
+                new BailEmailNotificationGenerator(
+                        newArrayList(adminOfficerBailSummaryUploadedPersonalisation,
+                                legalRepresentativeBailSummaryUploadedPersonalisation),
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/BailNotificationGeneratorConfiguration.java:46]: Did you test the scenario when there in No LR, does this Personalisation get triggered?

File: src/test/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/bail/adminofficer/AdminOfficerBailSummaryUploadedPersonalisationTest.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    private String homeOfficeReferenceNumber = "someHomeOfficeReferenceNumber";
+    private String applicantGivenNames = "someApplicantGivenNames";
+    private String applicantFamilyName = "someApplicantFamilyName";
+
+    @Mock
+    private EmailAddressFinder emailAddressFinder;
+    private AdminOfficerBailSummaryUploadedPersonalisation adminOfficerBailSummaryUploadedPersonalisation;
+
+    @BeforeEach
+    public void setup() {
```
Comment (neha-aggarw) [inline src/test/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/bail/adminofficer/AdminOfficerBailSummaryUploadedPersonalisationTest.java:43]: Please add a test for the scenario when there is no LR. It should pick appropriate template and not set the legalRepReference field.

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/BailNotificationHandlerConfiguration.java [hunk 6]
```diff
@@ -34,6 +36,38 @@ public PreSubmitCallbackHandler<BailCase> submitApplicationHearingCentreNotifica
         );
     }
 
+    @Bean
+    public PreSubmitCallbackHandler<BailCase> uploadSummaryNotificationHandler(
+            @Qualifier("uploadSummaryNotificationGenerator") List<BailNotificationGenerator> bailNotificationGenerators
+    ) {
+        return new BailNotificationHandler(
+                (callbackStage, callback) -> {
+                    BailCase bailCase = callback.getCaseDetails().getCaseData();
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/BailNotificationHandlerConfiguration.java:45]: Add a check in there:
 boolean isAllowedBailCase = (callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
                            && callback.getEvent() == Event.UPLOAD_BAIL_SUMMARY);
if (isAllowedBailCase) {
       BailCase bailCase = callback.getCaseDetails().getCaseData();
      return (callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
                            && callback.getEvent() == Event.UPLOAD_BAIL_SUMMARY
                            && isLegallyRepresented(bailCase));
} else 
return isAllowedBailCase;

**Review by neha-aggarw — APPROVED** [review]

**Review by rajeshthuraiyur — APPROVED** [review]
LGTM

**Pull request merged**